lib/active_merchant/billing/gateways/spreedly_core.rb

```diff
@@ -244,11 +244,20 @@ def parse(xml)
       end
 
       def childnode_to_response(response, node, childnode)
-        name = "#{node.name.downcase}_#{childnode.name.downcase}"
-        if name == 'payment_method_data' && !childnode.elements.empty?
-          response[name.to_sym] = Hash.from_xml(childnode.to_s).values.first
+        node_name = node.name.downcase
+        childnode_name = childnode.name.downcase
+        composed_name = "#{node_name}_#{childnode_name}"
+
+        childnodes_present = !childnode.elements.empty?
+
+        if childnodes_present && composed_name == 'payment_method_data'
+          response[composed_name.to_sym] = Hash.from_xml(childnode.to_s).values.first
+        elsif childnodes_present && node_name == 'gateway_specific_response_fields'
```

Comment:
Hi @curiousepic would you take a look here?
This is to enabled additional `gateway_specific_response_fields` which isn't returning by default in Active Merchant unlike the [direct Spree Gateway integration](https://docs.spreedly.com/payment-gateways/migs/#gateway-specific-response-fields).